website/static/js/markdown.js

```diff
@@ -30,12 +31,26 @@ var bootstrapTable = function(md) {
 var oldMarkdownList = function(md) {
     md.block.ruler.after('hr', 'pyMarkdownList', pymarkdownList);
 };
+var mfrURL = window.contextVars.mfrURL;
+var osfURL = window.contextVars.osfURL;
+
+var getMfrUrl = function (guid) {
+    return mfrURL + 'render?url='+ osfURL + guid + '/?action=download%26mode=render';
+};
 
 // Full markdown renderer for views / wiki pages / pauses between typing
 var markdown = new MarkdownIt('commonmark', {
     highlight: highlighter,
     linkify: true
-})
+    }).use(require('markdown-it-atrules'), {
+        type: 'osf',
+        pattern: /^http(?:s?):\/\/(?:www\.)?[a-zA-Z0-9 .:]{1,}\/render\?url=http(?:s?):\/\/[a-zA-Z0-9 .:]{1,}\/([a-zA-Z0-9]{1,})\/\?action=download|(^[a-zA-Z0-9]{1,}$)/,
+        format: function(assetID) {
+          var id = '__markdown-it-atrules-' + (new Date()).getTime() + Math.random() ;
```

Comment:
Does it make a difference if you use `Date.now()` instead of `(new Date()).getTime())`?

Reply:
I'd like to avoid `Math.random()` if possible. But if `Math.now()` doesn't fix the issue, just make sure to append a random integer instead of a random float. We don't want to have `.` characters in CSS IDs because they break CSS selectors (e.g. `#__markdown-it-atrules-1234.56` would select an element with ID `#__markdown-it-atrules-1234` and class `56`).

Reply:
I've tried `now()` and used the moment.js library The problem seems to be that every time something is added the div is redrawn fast enough that they're the same to the millisecond.